qemu/tests/block_scsi_device.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test the scsi device inside guest.
+
+    Scenario: from_delete_to_scan
+        1. Boot guest with a scsi data disk.
+        2. Delete the scsi data disk inside guest.
+        3. Check the scsi data disk inside guest.
+        4. Scan the scsi data disk inside guest.
+        5. Check the scsi data disk inside guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def scan_scsi_device(scsi_addr):
+        """ Scan the scsi device. """
+        error_context.context(
+            "Scan the scsi driver @%s." % scsi_addr, logging.info)
+        session.cmd('echo "- - -" > /sys/class/scsi_host/host%s/scan' %
+                    scsi_addr.split(':')[0])
+
+    def delete_scsi_device(scsi_addr):
+        """ Delete the scsi drive. """
+        error_context.context(
+            "Delete the scsi driver @%s." % scsi_addr, logging.info)
+        session.cmd('echo 1 > /sys/class/scsi_device/%s/device/delete' % scsi_addr)
+
+    def get_scsi_addr_by_product(product_name):
+        """ Get the scsi address by virtio_scsi product option. """
+        logging.info("Get the scsi address by qemu product option.")
+        addr = session.cmd(
+            "lsscsi | grep %s | awk '{print $1}'" % product_name).strip('[] ')
+        logging.info(
+            "The scsi address of the product %s is %s." % (product_name, addr))
+        return addr
```

Comment:
Failed to get addr. logs:

5:53:55 INFO | The scsi address of the product "AUTO HARDDISK" is 6:0:1:0]
.
05:53:55 INFO | Context: Delete the scsi driver @6:0:1:0]
.
05:53:55 DEBUG| Sending command: echo 1 > /sys/class/scsi_device/6:0:1:0]
/device/delete

Reply:
@XueqiangWei , thanks,  have updated it.